### ISSUE TITLE
null fields populated as "None" (string) on AWS account creation

### ISSUE THREAD

**Issue by lucab** [Netflix/security_monkey#133]
For some reason, when adding new accounts, null string fields get populated as 'None' (string).
You can see it when adding a new AWS account in the Setting panel.

When leaving some string fields empty, this is the request:
```
POST https://host/api/1/accounts

{"id":null,"active":true,"third_party":false,"name":"test","s3_name":"","number":"123","notes":null,"role_name":null}
```

and this is the reply:

```
{"third_party": false, "role_name": "None", "number": "123", "name": "test", "active": true, "notes": "None", "id": 49, "s3_name": "", "auth": {"authenticated": true, "user": "user"}}
```

**Comment by monkeysecurity** [comment]
This must be the dart code.  The API is receiving the String "None", as you described.

**Comment by lucab** [comment]
I had some time to track this further, and it looks like an issue in how we use flask.reqparse.
Basically, the field is really passed as `"notes": null` in the JSON. 
It is then converted to a `None` in the internal python representation.
Then, as this parameter is marked `type=unicode`, it is passed to `six.text_type()` and the conversion succeeds yielding a `'None'` string. So we end up with lots of "None" in the DB.

To me, it looks like a bug in reqparse, not yet reported.
However I'd like to first hotfix this here to unblock #120, and then discuss a proper fix in the flask module.

Comments?

**Comment by gene1wood** [comment]
@lucab hotfixing here in advance of working on the flask.reqparse upstream issue sounds good to me. What would the hotfix look like? (I'm in dire need of the functionality in your PR #120)

**Comment by lucab** [comment]
Something along the line of lucab@0c77861 (ie. a simple replacement from "None" to None after parsing the request).

**Comment by lucab** [comment]
Uhm, this seems to be flask-restful/flask-restful#390, which is fixed in Flask-RESTful 0.3.3. I just updated to it, and it seems to work greatly without my dirty patch.

**Comment by monkeysecurity** [comment]
Just to confirm, you believe this issue is fixed by upgrading flask-restful to 0.3.3?  No other action is required?

**Comment by lucab** [comment]
Yes, just upgrading the requirements fixes it, without further changes.
I have queued the commit in PR #120 just to avoid further conflict/rebases, but I can split it out if you prefer (or you can just cherry-pick it).

**Comment by monkeysecurity** [comment]
Fixed with #120.
